graphql-server/src/utils/cache.ts

```diff
@@ -0,0 +1,257 @@
+  const allSamplesQueryBody = buildSamplesQueryBody({
+    searchVals: [],
+    context: undefined,
+    filters: undefined,
+    addlOncotreeCodes: [],
+  });
+  const wesSamplesQueryBody = buildSamplesQueryBody({
+    searchVals: [],
+    context: WES_SAMPLE_CONTEXT,
+    filters: undefined,
+    addlOncotreeCodes: [],
+  });
+
+  // Run these queries concurrently to reduce server startup time
+  const oncotreeCache = inMemoryCache.get(ONCOTREE_CACHE_KEY) as OncotreeCache;
+  const allSamplesQueryPromises = buildSamplesQueryPromises({
+    queryBody: allSamplesQueryBody,
+    oncotreeCache,
+  });
+  const wesSamplesQueryPromises = buildSamplesQueryPromises({
+    queryBody: wesSamplesQueryBody,
+    oncotreeCache,
+  });
+  const queryResults = await Promise.all([
+    ...allSamplesQueryPromises,
+    ...wesSamplesQueryPromises,
+  ]);
+
+  // Add all samples query results to cache
+  const samplesCache: SamplesCache = {};
+  queryResults.forEach((result) => {
+    if (result) {
+      Object.assign(samplesCache, result);
+    }
+  });
+  inMemoryCache.set(SAMPLES_CACHE_KEY, samplesCache, SAMPLES_CACHE_TTL);
+}
+
+function buildSamplesQueryPromises({
+  queryBody,
+  oncotreeCache,
+}: {
+  queryBody: string;
+  oncotreeCache: OncotreeCache;
+}) {
+  const pageIndicesToCache = [...Array(SAMPLES_PAGES_TO_CACHE).keys()]; // [0, 1, 2, ..., n]
+  return pageIndicesToCache.map(async (pageIndex) => {
+    const samplesCypherQuery = await buildSamplesQueryFinal({
+      queryBody,
+      sort: SAMPLES_DEFAULT_SORT,
+      limit: CACHE_BLOCK_SIZE,
+      offset: CACHE_BLOCK_SIZE * pageIndex,
+    });
+    const queryResult = await queryDashboardSamples({
+      samplesCypherQuery,
+      oncotreeCache,
+    });
+    return queryResult ? { [samplesCypherQuery]: queryResult } : null;
+  });
+}
+
+export function updateCacheWithNewSampleUpdates(
+  newDashboardSamples: DashboardSampleInput[],
+  inMemoryCache: NodeCache
+) {
+  const samplesCache = inMemoryCache.get(SAMPLES_CACHE_KEY) as SamplesCache;
+
+  // Create a map out of newDashboardSamples for quick lookup by primaryId
+  const newSamplesByPrimaryId = newDashboardSamples.reduce(
+    (accumulator, newDashboardSample) => {
+      accumulator[newDashboardSample.primaryId] = newDashboardSample;
+      return accumulator;
+    },
+    {} as Record<string, DashboardSampleInput> // key = primaryId
+  );
+
+  // Update the fields of any samples in cache that were changed in newDashboardSamples
+  for (const sample of Object.values(samplesCache).flat()) {
+    if (newSamplesByPrimaryId.hasOwnProperty(sample.primaryId)) {
+      const newDashboardSample = newSamplesByPrimaryId[sample.primaryId];
+      for (const field of newDashboardSample.changedFieldNames) {
+        if (field in sample && field in newDashboardSample) {
+          (sample as any)[field] =
+            newDashboardSample[field as keyof DashboardSampleInput];
```

Comment:
I casted `sample`'s type to `any` because Typescript couldn't easily infer that `field` exists inside `sample`, even though this is inside an if statement that checks that `field` is in `sample`. This isn't the cleanest Typescript approach but it's safe.